geomstats/geometry/general_linear.py

```diff
@@ -159,3 +165,57 @@ def path(time):
             return cls.exp(vecs, base_point)
 
         return path
+
+
+class GeneralLinearLieAlgebra(MatrixLieAlgebra):
```

Comment:
This is the Lie algebra of the general linear group, thus the space of matrices. 
Why isn't `Matrices(MatrixLieAlgebra)` enough, i.e. why not adding the functions into `Matrices` directly?

Reply:
We usually do not name classes in terms of whose Lie algebra they are, but in terms of what they are, thus GeneralLinearLieAlgebra feels a bit weird there?

Reply:
I see... But then:
- we have two spaces that are the same but have different names... (Matrices and GeneralLinearAlgebra)
- the fact that a space is the linear algebra of another is usually contained in the attribute: space.lie_algebra, not in the name. 
  - I did not see SpecialLinearLieAlgebra yet: is it in @lpereira95 's PR? I would argue against naming the manifolds this way.
  - Otherwise, we would have two logics to name Lie algebras: either by the manifold they actually represent (skewsym matrices for example) or by the space of whom they are a Lie algebra. Which can become confusing.

Is there another way we could make this work with the inheritances? :/ How about SquareMatrices(MatrixLieAlgebra)?